support-frontend/app/controllers/RedemptionController.scala

```diff
@@ -121,10 +122,14 @@ class RedemptionController(
   def displayProcessing(redemptionCode: String): Action[AnyContent] =
     authenticatedAction(subscriptionsClientId).async {
       implicit request: AuthRequest[Any] =>
-        val processingPage = for {
+        val processingPage: EitherT[Future, String, Result] = for {
           user <- identityService.getUser(request.user.minimalUser)
           corporateCustomer <- getCorporateCustomer(redemptionCode)
-        } yield showProcessing(redemptionCode, corporateCustomer, user)
+          hasDigipack <- EitherT.right(userHasDigitalSubscription(membersDataService, request.user))
```

Comment:
will be ok, but we could do the check first, then we don't depend on the calls to identity and dynamo

Reply:
Good suggestion 👍